initializr-web/src/main/java/io/spring/initializr/web/project/ProjectRequestToDescriptionConverter.java

```diff
@@ -68,16 +69,57 @@ public ProjectDescription convert(ProjectRequest request,
 		description.setPlatformVersion(Version.parse(springBootVersion));
 		description.setVersion(determineValue(request.getVersion(),
 				() -> metadata.getVersion().getContent()));
+		conventionCorrection(description);
 		resolvedDependencies
 				.forEach((dependency) -> description.addDependency(dependency.getId(),
 						MetadataBuildItemMapper.toDependency(dependency)));
+
 		return description;
 	}
 
 	private String determineValue(String candidate, Supplier<String> fallback) {
 		return (StringUtils.hasText(candidate)) ? candidate : fallback.get();
 	}
 
+	private void conventionCorrection(ProjectDescription description) {
+		baseDirectoryCorrection(description);
+		artificialIdCorrection(description);
+		groupIdCorrection(description);
+	}
+
+	private void baseDirectoryCorrection(ProjectDescription description) {
+		if (description.getBaseDirectory() != null
+				&& description.getBaseDirectory().equals(description.getArtifactId())) {
+			description.setBaseDirectory(
+					coordinateCorrection(description.getBaseDirectory()));
+		}
+	}
+
+	private void artificialIdCorrection(ProjectDescription description) {
+		description.setArtifactId(coordinateCorrection(description.getArtifactId()));
+	}
+
+	private void groupIdCorrection(ProjectDescription description) {
+		description.setGroupId(coordinateCorrection(description.getGroupId()));
+	}
+
+	private String coordinateCorrection(String coordinate) {
```

Comment:
If the value doesn't contain any invalid characters, we don't need to do any of this and can just return it as is.